src/Table.jl

```diff
@@ -304,3 +304,12 @@ end
 function Base.vec(t::Table{<:NamedTuple{names}}) where {names}
     return Table(map(vec, columns(t)))
 end
+
+# Specialize for empty table of type `Union{}`
+const EmptyTable{N} = Table{Union{}, N, Nothing}
+Base.empty(::Table{<:Any, 1}, ::Type{Union{}}) = EmptyTable{1}(nothing)
```

Comment:
What I had in mind was something like

```suggestion
Base.empty(::Type{<:Table{<:Any, 1}}) = EmptyTable{1}(nothing)
```

i.e., I want an empty table given its type. I also would like to suggest omitting the eltype since this is _the_ only eltype that is guaranteed to be able to widen to anything.

Reply:
Oh you want to use types not instances? I’ve been heavily trending the other direction lately.

And the last part - you mean omit the eltype in `empty`’a function signature and default to `Union{}`, not the `eltype` of the input?

Reply:
For the first part, please see my comment in https://github.com/JuliaData/TypedTables.jl/issues/55#issuecomment-552155319

> you mean omit the eltype in `empty`’s function signature and default to `Union{}`, not the `eltype` of the input?

I was suggesting to make the second argument optional.  So, e.g., returning `Union{}[]` from `empty(::Type{Vector})`.  Though `empty(::Type{Vector{Float64}})` probably still should return `Float64[]`.